File: python/easy/minimum_index_sum_of_two_lists.py
class Solution(object):
    def findRestaurant(self, list1, list2):
        """
        :type list1: List[str]
        :type list2: List[str]
        :rtype: List[str]
        """
        d1 = dict()
        least_idx_sum = sys.maxint
        idx = 0
        ret = list()
        for idx in range(0, len(list2)):
            d1[list2[idx]] = idx
        idx = 0
        for idx in range(0, len(list1)):
            if list1[idx] in d1:
                sum = idx + d1[list1[idx]]
                if sum == least_idx_sum:
                    ret.append(list1[idx])
                elif sum < least_idx_sum:
                    least_idx_sum = sum
                    ret= []
                    ret.append(list1[idx])
        return ret

File: cpp/easy/minimum_index_sum_of_two_lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        unordered_map<string, int> my_map;
        vector<string> ret;
        for(int i = 0; i < list2.size(); ++i)
            my_map[list2[i]] = i;
        int least_index_sum = INT_MAX;
        for(int j = 0; j < list1.size(); ++j)
        {
            int sum = 0;
            if(my_map.find(list1[j]) != my_map.end())
            {
                sum = j + my_map[list1[j]];
                if(least_index_sum == sum)
                    ret.push_back(list1[j]);
                if(least_index_sum > sum)
                {
                    least_index_sum = sum;
                    ret.clear();
                    ret.push_back(list1[j]);
                }
            }
        }
        return ret;
    }
};


Conclustion:
hash table, so easy... no extra comments.
